metrics.go

```diff
@@ -10,38 +10,47 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-// Metrics store Hub collected metrics.
-type Metrics struct {
+type Metrics interface {
+	// ConnectSubscriber collects metrics about about subscriber connections.
+	ConnectSubscriber(s *Subscriber)
```

Comment:
From a semantic point of view, I think these are events you want to subscribe to, so `SubscriberConnected`, `SubscriberDisconnected` and `UpdatePublished` sound better to me.

Reply:
I was hesitating between the two forms, so let's do this, thanks.